app-config-webpack/src/index.ts

```diff
@@ -87,28 +86,40 @@ export default class AppConfigPlugin implements Options {
 
   injectHead(compiler: Compiler) {
     compiler.hooks.compilation.tap('AppConfigPlugin', (compilation) => {
-      HtmlWebpackPlugin.getHooks(compilation).alterAssetTagGroups.tapPromise(
-        'AppConfigPlugin',
-        async ({ headTags, ...html }) => {
-          // remove placeholder <script id="app-config"></script> if it exists
-          const newTags: HtmlTagObject[] = headTags.filter(
-            ({ attributes }) => attributes.id !== 'app-config',
+      import('html-webpack-plugin')
+        .then((module) => {
+          const HtmlWebpackPlugin = module.default;
+
+          HtmlWebpackPlugin.getHooks(compilation).alterAssetTagGroups.tapPromise(
+            'AppConfigPlugin',
+            async ({ headTags, ...html }) => {
+              // remove placeholder <script id="app-config"></script> if it exists
+              const newTags = headTags.filter(({ attributes }) => attributes.id !== 'app-config');
+
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-call
```

Comment:
@joelgallant Having issues importing the `HtmlTagObject` type from a lazy loaded module. Any suggestions?

Adding this:
```js
.then((module) => {
  const HtmlWebpackPlugin = module.default;
  const { HtmlTagObject } = module;
```
Results in these errors:
```
'HtmlTagObject' is declared but its value is never read.ts(6133)
Property 'HtmlTagObject' does not exist on type '{ default: typeof HtmlWebpackPlugin; prototype: HtmlWebpackPlugin; version: number; }'.ts(2339)
'HtmlTagObject' is assigned a value but never used. Allowed unused vars must match /^_+$/u.eslint[@typescript-eslint/no-unused-vars](https://github.com/typescript-eslint/typescript-eslint/blob/v3.10.1/packages/eslint-plugin/docs/rules/no-unused-vars.md)
```

Reply:
`HtmlTagObject` is a type only. You can `import type { HtmlTagObject } from 'html-webpack-plugin'` up top. That won't get included in the bundle - TS erases it at runtime.

Reply:
Ah perfect. Thank you